comp/dogstatsd/server/server.go

```diff
@@ -371,12 +375,14 @@ func (s *server) start(context.Context) error {
 	}
 
 	if len(socketPath) > 0 {
-		unixListener, err := listeners.NewUDSDatagramListener(packetsChannel, sharedPacketPoolManager, sharedUDSOobPoolManager, s.config, s.tCapture, s.wmeta, s.pidMap, s.listernersTelemetry, s.packetsTelemetry, s.telemetry)
-		if err != nil {
-			s.log.Errorf("Can't init listener: %s", err.Error())
-		} else {
-			tmpListeners = append(tmpListeners, unixListener)
-			udsListenerRunning = true
+		if _, err := os.Stat(filepath.Dir(socketPath)); !errors.Is(err, fs.ErrNotExist) {
```

Comment:
Do you think we can remove this entire condition and restore logic the way it was originally? We should always report something if we couldn't create the socket, both for default and user-provided paths.

Reply:
Ok, re-reading the code if the parent directory doesn't exist it will just log `"Can't init UDS listener on path %s: %s"`, and continue the server start (no panic). This should be fine regardless of where the value is coming from (default vs explicitly set).

Also, just to clarify, by "restore the logic the way it was originally" you are referring exactly to the point above, not to the original PR where I had a separate listener for the default socket, right?